### PR TITLE
Shreeya_task_07

### PR DESCRIPTION
**Pull request by shreeya-tk**
High score: 17159 

breakoutgame_screenshort

### CODE REVIEW

File: Exercises/Shreeya, Tulabandula/task_07.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+        self.grid = {}  
+        self.score = 0
+        self.ball_position = (0, 0)
+        self.paddle_position = (0, 0)
+
+    def update_grid(self, output):
+        for i in range(0, len(output), 3):
+            x, y, tile_type = output[i:i + 3]
+            if x == -1 and y == 0: 
+                self.score = tile_type
```
Comment (HannahGedler) [inline Exercises/Shreeya, Tulabandula/task_07.py:133]: when I run the program I can just see half of the score, but maybe this is just a problem with the layout on my  laptop

File: Exercises/Shreeya, Tulabandula/task_07.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+            else:
+                self.grid[(x, y)] = tile_type
+                if tile_type == 4:  
+                    self.ball_position = (x, y)
+                elif tile_type == 3: 
+                    self.paddle_position = (x, y)
+
```
Comment (HannahGedler) [inline Exercises/Shreeya, Tulabandula/task_07.py:140]: this is a nice way to set the coordinates

File: Exercises/Shreeya, Tulabandula/task_07.py [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+        return 0  
+
+    def render(self):
+        max_x = max(pos[0] for pos in self.grid)
+        max_y = max(pos[1] for pos in self.grid)
+        grid_array = np.zeros((max_y + 1, max_x + 1), dtype=int)
+
+        for (x, y), tile_type in self.grid.items():
+            grid_array[y, x] = tile_type
+
```
Comment (HannahGedler) [inline Exercises/Shreeya, Tulabandula/task_07.py:157]: I really like the idea of this function and how it works

File: Exercises/Shreeya, Tulabandula/task_07.py [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+            color_grid = np.zeros((new_grid.shape[0], new_grid.shape[1], 3))
+            for tile_type, color_name in TILE_COLORS.items():
+                color_grid[new_grid == tile_type] = to_rgb(color_name)
+            img.set_data(color_grid)
+
+            score_text.set_text(f"Score: {game.score}")
+
+            return [img, score_text]
+
+    ani = FuncAnimation(fig, play_game, interval=50, blit=True)
```
Comment (HannahGedler) [inline Exercises/Shreeya, Tulabandula/task_07.py:197]: I did not know about this function, it fits really well here

File: Exercises/Shreeya, Tulabandula/task_07.py [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+#
+# BONUS: (no extra points, just for fun)
+# make a movie of playing the game :)
+
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.animation import FuncAnimation
+from matplotlib.colors import to_rgb
+
+class IntComputer:
```
Comment (PixelWelt) [inline Exercises/Shreeya, Tulabandula/task_07.py:52]: great idea putting the whole Int Computer in a class

File: Exercises/Shreeya, Tulabandula/task_07.py [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+    def update_grid(self, output):
+        for i in range(0, len(output), 3):
+            x, y, tile_type = output[i:i + 3]
+            if x == -1 and y == 0: 
+                self.score = tile_type
+            else:
+                self.grid[(x, y)] = tile_type
+                if tile_type == 4:  
+                    self.ball_position = (x, y)
+                elif tile_type == 3:
```
Comment (PixelWelt) [inline Exercises/Shreeya, Tulabandula/task_07.py:138]: using an enum instead of plain numbers could make your code better to understand

**Review by PixelWelt — APPROVED** [review]
Please try using flake8 in VS Code to prevent pep8 errors. I did not mark them, but there are some. 

All in all I really think, that you have one of the best implementations of the game!

**Pull request merged**